include/h2o.h

```diff
@@ -618,8 +618,18 @@ typedef struct st_h2o_generator_t {
     void (*stop)(struct st_h2o_generator_t *self, h2o_req_t *req);
 } h2o_generator_t;
 
-typedef int (*h2o_ostream_pull_cb)(h2o_generator_t *generator, h2o_req_t *req, h2o_iovec_t *buf);
+typedef enum h2o_send_state {
+    H2O_SEND_STATE_IN_PROGRESS,
+    H2O_SEND_STATE_FINAL,
+    H2O_SEND_STATE_ERROR,
+} h2o_send_state_t;
 
+typedef h2o_send_state_t (*h2o_ostream_pull_cb)(h2o_generator_t *generator, h2o_req_t *req, h2o_iovec_t *buf);
+
+static inline int h2o_stream_send_state_is_final(h2o_send_state_t s)
```

Comment:
Could you please consider introducing `h2o_send_state_is_in_progress` instead of `is_final`?

The reason I ask is because current name is confusing; the function is named `is_final` even though it returns true if the state is either final _or error_.

I might have not been clear enough about this point in https://github.com/h2o/h2o/pull/1031#discussion_r75584249, sorry for the confusion.

PS. And thank you for changing all the parts of the code to tri-state (as well as renaming the variable to avoid confusion). I highly appreciate that.